web/src/settings_org.js

```diff
@@ -824,6 +842,43 @@ export function init_dropdown_widgets() {
     });
     settings_components.set_can_access_all_users_group_widget(can_access_all_users_group_widget);
     can_access_all_users_group_widget.setup();
+
+    const resolve_topic_group_widget = new dropdown_widget.DropdownWidget({
+        widget_name: "realm_resolve_topic_group",
+        get_options: () =>
+            user_groups.get_realm_user_groups_for_dropdown_list_widget(
+                "resolve_topic_group",
+                "realm",
+            ),
+        $events_container: $("#settings_overlay_container #organization-permissions"),
+        item_click_callback(event, dropdown) {
+            dropdown.hide();
+            event.preventDefault();
+            event.stopPropagation();
+            settings_components.resolve_topic_group_widget.render();
+            settings_components.save_discard_widget_status_handler($("#org-moving-msgs"));
+        },
+        tippy_props: {
+            placement: "bottom-start",
+        },
+        default_id: get_resolve_topic_widget_option(),
+        on_mount_callback(dropdown) {
+            $(dropdown.popper).css("min-width", "300px");
+        },
+        unique_id_type: dropdown_widget.DataTypes.NUMBER,
+    });
+    settings_components.set_resolve_topic_group_widget(resolve_topic_group_widget);
+    resolve_topic_group_widget.setup();
+}
+
+function get_resolve_topic_widget_option() {
+    if (
+        realm.realm_resolve_topic_group === user_groups.get_user_group_from_name("role:nobody").id
+    ) {
+        update_resolve_topic_sub_settings(false);
```

Comment:
When initial group of the `resolve_topic_group` is "role:nobody", we don't want to show it in dropdown option. Hence, we rather show the checkbox as unchecked and dropdown with "role:members" (default).

Reply:
>We can just show the option which was selected before the setting was disabled and I guess this is what we do for other similar settings.

The problem here is that we are using just one database field for both `allow_resolve_topic` and `can_resolve_topics_group` settings. So technically, if you uncheck the checkbox, the value of this field is `role:nobody` -- which removes the possibility of storing the last option of dropdown. Hence, I thought I would just simply show the default value (members).